data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,598 @@
+    """
+    Identify, log and remove rows with suppressed concepts
+
+    Note: Rows are appended to log and backup tables if they already exist
+    
+    :param project_id: Identifies the project containing all associated datasets
+    :param sandbox_dataset_id: Identifies a dataset to store log and backup records
+    :param concept_lookup_table: Identifies the table containing the concepts to suppress
+    :param target_dataset_ids: List of datasets to retract data from
+    
+    """
+    # get suppress row info
+    rows_dest_table = f'{sandbox_dataset_id}.{ROWS_RESOURCE_NAME}'
+    tables_to_suppress_df = get_tables_to_suppress_df(project_id,
+                                                      target_dataset_ids)
+    suppress_rows_df = get_rows_to_suppress_df(tables_to_suppress_df,
+                                               concept_lookup_table)
+    suppress_rows_df.to_csv(f'{ROWS_RESOURCE_NAME}.csv')
+    suppress_rows_df.to_gbq(destination_table=rows_dest_table,
+                            if_exists='append')
+
+    # get suppress summary
+    summary_dest_table = f'{sandbox_dataset_id}.{SUMMARY_RESOURCE_NAME}'
+    suppress_summary_df = get_suppress_summary_df(suppress_rows_df)
+    suppress_summary_df.to_csv(f'{SUMMARY_RESOURCE_NAME}.csv')
+    suppress_summary_df.reset_index(inplace=True)
+    suppress_summary_df.to_gbq(destination_table=summary_dest_table,
+                               if_exists='append')
+
+    # backup rows
+    backup_rows_to_suppress(tables_to_suppress_df, rows_dest_table)
+
+    # delete rows
+    all_delete_queries = get_delete_queries(rows_dest_table)
+    delete_query_jobs = run_delete_queries(all_delete_queries)
+    print_jobs(delete_query_jobs)
+
+
+def get_arg_parser():
+    """
+    Get the parser for command line arguments
+    :return: the parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-p',
+        '--project_id',
+        action='store',
+        dest='project_id',
+        help='Identifies the project containing the datasets to retract from',
+        required=True)
+    subparsers = parser.add_subparsers(dest='cmd')
+    setup_parser = subparsers.add_parser('setup')
+    setup_parser.add_argument(
+        'concept_lookup_dest_table',
+        action='store',
+        help='Table where suppressed concepts should be loaded')
+    retract_parser = subparsers.add_parser('retract')
+    retract_parser.add_argument(
+        '-s',
+        '--sandbox_dataset_id',
+        action='store',
+        dest='sandbox_dataset_id',
+        help='Identifies the dataset where output is stored',
+        required=True)
+
+    retract_parser.add_argument(
+        '-c',
+        '--concept_lookup_table',
+        action='store',
+        dest='concept_lookup_table',
+        help='Table where suppressed concepts are loaded',
+        required=True)
+
+    retract_parser.add_argument('-d',
+                                '--dataset_ids',
```

Comment:
Why the multiple parsers?  I feel like I'm not understanding something.

Reply:
You're correct- I will add more descriptive comment.